rfc/rfc-56/rfc-56.md

```diff
@@ -0,0 +1,248 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-56: Early Conflict Detection For Multi-writer
+
+
+## Proposers
+
+- @zhangyue19921010
+
+## Approvers
+ - @yihua
+
+## Status
+
+JIRA: https://issues.apache.org/jira/browse/HUDI-1575
+
+
+## Abstract
+
+At present, Hudi implements an optimized OCC (Optimistic Concurrency Control) based on timeline to ensure data
+consistency, integrity and correctness between multi-writers. However, the related conflict detection is performed
+before commit metadata and after the data writing is completed. If this detection was failed, it would lead to a waste
```

Comment:
nit: "If this detection was failed" -> "if a conflict is detected"

I guess this makes it more clear.

Reply:
Changed. Thanks a lot for your review and attention :)